components/dashboard/src/provider-utils.tsx

```diff
@@ -75,12 +75,21 @@ async function openAuthorizeWindow(params: OpenAuthorizeWindowParams) {
 
         if (typeof event.data === "string" && event.data.startsWith("success")) {
             killAuthWindow();
-            onSuccess && onSuccess();
+            onSuccess && onSuccess(event.data);
         }
         if (typeof event.data === "string" && event.data.startsWith("error:")) {
-            const errorAsText = atob(event.data.substring("error:".length));
+            let error: string | { error: string, description?: string } = atob(event.data.substring("error:".length));
+            try {
+                const payload = JSON.parse(error);
```

Comment:
`JSON.parse` on user-provided input is a minefield in general. We should be careful to not introduce XSS issues here..

Reply:
We checked the usage and deemed it fine. The issues basically is if either:
  - you have an old browser with an polyfill `JSON.parse` -> `eval`
  - you set user-provided input to `href`/`src` attributes without validation.
 We do not do any of this here.

Reply:
to summarize our discussion on that. 
`JSON.parse` isn't insecure per se, but we must pay attention to not just use parsed values as URLs for instance of `img.src`, etc. 

but even that is hard to reproduce and share, because the message event listener is only active when the auth window is open.